primitives/io/src/lib.rs

```diff
@@ -370,6 +373,13 @@ pub trait Misc {
 			.call_in_wasm(wasm, "Core_version", &[], &mut ext)
 			.ok()
 	}
+
+	/// Get the current system time in nanoseconds.
```

Comment:
```suggestion
	/// Get the number of nanoseconds passed since the UNIX epoch.
```

Reply:
However, I wonder why we use `SystemTime` here? System time is known to be non-monotonic and not consistent and usually is not used for benchmarking (which is the only purpose of this host function, right?). Can we use some other clock like uptime based or something similar?

Reply:
Well, @pepyakin, we know that using [`arch::_rdtsc`](https://doc.rust-lang.org/core/arch/x86_64/fn._rdtsc.html) has its own issues only working on certain architectures.

The question really boils down to: is `SystemTime` good enough? I think we will need to use it to know?

I dont think anyone should feel this commit is us committing to anything. I just want to make this function available so others like @mattrutherford can also start doing some benchmarking work.

Reply:
Agreed, we can iterate later